go-controller/pkg/ovn/base_network_controller_pods.go

```diff
@@ -538,8 +538,10 @@ func (bnc *BaseNetworkController) addLogicalPortToNetwork(pod *kapi.Pod, nadName
 	// it for the default network as well. If at all possible, keep them
 	// functionally equivalent going forward.
 	var annotationUpdated bool
-	if bnc.IsSecondary() {
+	if bnc.IsSecondary() || bnc.IsPrimaryNetwork() {
```

Comment:
might want to revisit the definition of a primary net; right now, a user defined primary network *is* a secondary net. So this added or condition is **not** required.